### ISSUE TITLE
How do I preserve the background and fix the mask position?

### ISSUE THREAD

**Issue by Chaejongwook1** [eladrich/pixel2style2pixel#293]
Hello. Thank you for providing a good source code.

I am a university student who is doing research in Korea.
Currently, I want to create a polyp in the mask part by synthesizing a mask in the image with pixel2style2pixel.
Below are the training result and the inference result.

training result
![image](https://user-images.githubusercontent.com/65751069/195500752-3bf6ed18-ad4f-43db-8144-cbd9f620fdbf.png)
![image](https://user-images.githubusercontent.com/65751069/195500807-b666c7ab-3598-46d6-8dad-1b66b09a3091.png)
![image](https://user-images.githubusercontent.com/65751069/195500836-d515a36f-a4eb-4c08-be39-66c3486ede3a.png)

inference result
![image](https://user-images.githubusercontent.com/65751069/195500965-d3cae1de-6cf5-4a04-ab0c-362a80846dd7.png)
![image](https://user-images.githubusercontent.com/65751069/195500986-ad06473e-f043-461f-a2eb-a128d9bddc87.png)
![image](https://user-images.githubusercontent.com/65751069/195501250-8d111ca2-b723-44d9-a7d5-6dcd5ac60717.png)

What I want here is to preserve the background as it is and create a polyp in the mask, but it's not being done properly.
What I want the most is to produce a polyp exactly at the location of the mask.

Is it impossible to work with the model I want? If possible, I would appreciate it if you could advise me in which direction to proceed with the learning.

# polyp_inpainting 2022-09-26 
!python scripts/train.py \
--dataset_type=ffhq_encode \
--exp_dir=experiment/{date()} \
--workers=8 \
--batch_size=16 \
--test_batch_size=16 \
--test_workers=8 \
--val_interval=2500 \
--save_interval=5000 \
--encoder_type=GradualStyleEncoder \
--start_from_latent_avg \
--lpips_lambda=0.8 \
--l2_lambda=2 \
--moco_lambda=0.2 \
--use_wandb \
--output_size=256 \
--stylegan_weights="pretrained_models/network-snapshot-000800.pt"

**Comment by yuval-alaluf** [comment]
If you are seeing that the background is not preserved, you can create the final output image by taking the non-masked region from the original image and the masked region from your generated result. Then, you will need to do some post-processing to merge the two regions together as they will not look natural if you combine them naively. You could try applying a poison blending or some soft mask blending between the two regions.
Hope this helps.

**Comment by Chaejongwook1** [comment]
Thank you for your answer.
Is there a way to induce polyps through hyperparameters so that they are accurately produced in the mask position even if the quality is poor? Since the mask position should be used as Ground Truth, we want a polyp to be generated accurately in the white part even if the quality is poor.

**Comment by yuval-alaluf** [comment]
You could possibly try to strengthen the loss within the masked region. This could force the encoder to focus more on the area of interest in your case.
